.github/workflows/nightly-website.yml

```diff
@@ -71,3 +74,41 @@ jobs:
           git add --force dev/
           git commit -m "publish documentation" --allow-empty
           git push origin asf-site:asf-site
+
+  publish-tag:
+    name: "Publish Tag Website"
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          # NOTE: needed to push at the end
+          persist-credentials: true
+          ref: asf-site
+      - name: Add tagged docs
+        uses: actions/download-artifact@v3
+        with:
+          name: docs
+          path: temp
+      - name: Determine Path
+        id: determine-path
+        run: |
+          regex='^refs/tags/adbc-([0-9]+\.[0-9]+\.[0-9]+)(-rc[0-9]+)?$'
+          if [[ ${{ github.ref }} =~ $regex ]]; then
+            mv temp ${BASH_REMATCH[1]}
+            git add --force ${BASH_REMATCH[1]}
+            ls -laR ${BASH_REMATCH[1]}
+          else
+            exit 1
+          fi
+      - name: Build
+        shell: bash
+        run: |
+      - name: Push changes to asf-site branch
+        run: |
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'actions@github.com'
```

Comment:
We can use https://avatars.githubusercontent.com/in/15368?s=40&v=4 icon with `github-actions[bot]@users.noreply.github.com`.

See also:
* https://github.com/apache/arrow-datafusion-python/pull/104#discussion_r1040341981
* https://github.com/apache/arrow-datafusion-python/commit/a1100499a94e2ab48ab852b8c76f3f9a2db59e01#diff-d54d69dbb27e75dae25cb4b2384310cb57707e419377cf572d5cb0ecc1f16877R71